lms/djangoapps/certificates/queue.py

```diff
@@ -306,6 +307,14 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
                 #   Despite blowing up the xml parser, bad values here are fine
                 grade_contents = None
 
+            # If this user's enrollment is not eligible to receive a
+            # certificate, mark it as such for reporting and
+            # analytics.
+            if not is_whitelisted and not CourseMode.is_eligible_for_certificate(enrollment_mode):
```

Comment:
I'm trying to think of a better way to structure this code, but nothing's really jumping out at me (other than removing the redundant check I mentioned). Do you have any suggestions?